kubeflow/trainer/backends/kubernetes/utils.py

```diff
@@ -49,6 +49,13 @@ def get_container_devices(
     elif constants.CPU_LABEL in resources.limits:
         device = constants.CPU_LABEL
         device_count = resources.limits[constants.CPU_LABEL].actual_instance
+    elif any(k.startswith(constants.GPU_MIG_PREFIX) for k in resources.limits):
+        mig_keys = [k for k in resources.limits if k.startswith(constants.GPU_MIG_PREFIX)]
+        if len(mig_keys) > 1:
+            raise Exception(f"Multiple MIG resource types are not supported yet: {mig_keys}")
+        mig_key = mig_keys[0]
+        device = mig_key.split("/")[1]
+        device_count = resources.limits[mig_key].actual_instance
```

Comment:
Shall we also allow to set MIG devices when user creates TrainJob: https://github.com/kubeflow/sdk/blob/5a956ab690ce011a924b8055cdd73d85a2309f84/kubeflow/trainer/backends/kubernetes/utils.py#L235-L236 ?

cc @kubeflow/kubeflow-sdk-team @kubeflow/kubeflow-trainer-team